src/Avatar.js

```diff
@@ -1,39 +1,38 @@
-import React from 'react'
 import PropTypes from 'prop-types'
-import styled from 'react-emotion'
-import {themeGet} from 'styled-system'
-import {withSystemProps} from './system-props'
-
-const Avatar = styled(AvatarImage)`
-  display: inline-block;
-  overflow: hidden; // Ensure page layout in Firefox should images fail to load
-  line-height: ${themeGet('lineHeights.condensedUltra', 1)};
-  vertical-align: middle;
-  ${borderRadius};
-`
+import styled from 'styled-components'
+import {themeGet, space} from 'styled-system'
+import theme from './theme'
 
 function borderRadius({size}) {
   return {
     borderRadius: size <= 24 ? '2px' : '3px'
   }
 }
 
-function AvatarImage({alt, size, ...rest}) {
-  return <img alt={alt} width={size} height={size} {...rest} />
-}
-
-// styled() changes this
-Avatar.displayName = 'Avatar'
+const Avatar = styled.img.attrs(props => ({
+  height: props.size,
+  width: props.size
+}))`
+  display: inline-block;
+  overflow: hidden; // Ensure page layout in Firefox should images fail to load
+  line-height: ${themeGet('lineHeights.condensedUltra', 1)};
```

Comment:
Another potential sidebenefit: it'd "fake deep-merge" a theme from props rather than replacing it whole-sale.

Imagine overriding the theme from props, e.g. `<Component theme={{ lineHeights: { condensedUltra: 1 } } }} />`.

If you're using `defaultProps`, you'd end up with the rest of theme being undefined and potentially lots of errors if nested properties are used in the component.

If you're using that util you'd end up with the rest still being defaulted to the default theme, no errors with a single override!

Reply:
Missed all these comments! I don't think we should force users to use a ThemeProvider. It would be nice if everyone did, and I hope that exporting a custom theme provider will encourage people to do so, but I like the idea of Primer/Components (& it's theme!) being able to be used out of the box without needing to config a ThemeProvider